src/item.cpp

```diff
@@ -1076,34 +1078,82 @@ std::string item::info(bool showtext, std::vector<iteminfo> &dump_ref) const
         it_tool* tool = dynamic_cast<it_tool*>(type);
 
         if ((tool->max_charges)!=0) {
-            std::string charges_line = _("Charges"); //;
+            int t_max;
```

Comment:
Declaring the variables `t_max`, `temp_fmt` etc. outside of the if-blocks where they are used makes the impressions there is some connection between the logic in the block and the code afterwards. I expected something like this:
```C++
int f_max;
// set f_max depending on various conditions:
if( condition1 ) {
    f_max = value1;
} else {
    f_max = value2;
}
print( "max: %d", f_max );
```

But here, there is not such connection, each block uses the variable separately. Therefor the variables should be declared in each block (and you can actually make them const there):
```C++
if( condition1 ) {
    const int f_max = value1;
    print( "max: %d", f_max );
} else {
    const int f_max = value2;
    print( "max: %d", f_max );
}
```

Or maybe just move the `dump->push_back(iteminfo(` line out of the blocks and at the end as it is common to all blocks.